macro/src/lib.rs

```diff
@@ -21,3 +22,28 @@ pub fn write_html(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
 pub fn derive_html_component(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
 	component::derive_html_component(input.into()).into()
 }
+
+/// makes a function that returns std::fmt::Result into a component function, argument to the
+/// attribute is the name of the struct that will be generated
+/// usage:
+/// ```rust
+/// #[component(MyComponent)]
+/// pub fn my_component(title: String) -> std::fmt::Result {
+///		component_html!(
+///			<div>{title}</div>
+///		)
+/// }
+/// ```
```

Comment:
```suggestion
/// Makes a function that returns `std::fmt::Result` into a component function, argument to the
/// attribute is the name of the struct that will be generated.
///
/// # Usage
///
/// ```
/// #[component(MyComponent)]
/// pub fn my_component(title: String) -> std::fmt::Result {
///     component_html!(
///         <div>{title}</div>
///     )
/// }
/// ```
```